torch/csrc/api/src/serialize/input-archive.cpp

```diff
@@ -19,6 +19,12 @@ namespace serialize {
 InputArchive::InputArchive()
     : module_(std::make_shared<jit::script::Module>()) {}
 
+bool InputArchive::has_key(const std::string& key) {
```

Comment:
Can we combine this logic with the first few lines of `InputArchive::read` ? Maybe by making an underlying `bool try_read(...)` and having `read` call that and error out if it returns false. I just don't like having the same logic in two places (and doing the same work twice in each iteration of your while loop in serialize.h)

Reply:
Fixed